### PR TITLE
feat: replace proto build #NTRN-326

### PR DESCRIPTION
**Pull request by joldie777**
Task: https://hadronlabs.atlassian.net/browse/NTRN-326

Related PRs:
https://github.com/neutron-org/neutron/pull/634
https://github.com/neutron-org/neutron-dev-contracts/pull/59
https://github.com/neutron-org/neutron-integration-tests/pull/329

List of changes:
1. Replaced proto build script.
2. Replaced `Stargate` with `Grpc` for `QueryRequest` and `Any` for `CosmosMsg`.
3. [Returned](https://github.com/neutron-org/neutron-sdk/blob/c89d40468bc2f06493b94f0ed058b28c794f574a/proto-build/src/main.rs#L131) generation of the `tendermint.types.EventDataRoundState` struct.
4. [Added](https://github.com/neutron-org/neutron-sdk/blob/c89d40468bc2f06493b94f0ed058b28c794f574a/proto-build/src/transform.rs#L79-L86) boxing of the large fields to reduce the total size of the enums.
5. [Removed](https://github.com/neutron-org/neutron-sdk/blob/c89d40468bc2f06493b94f0ed058b28c794f574a/proto-build/src/transform.rs#L114-L116) redundant `CosmwasmExt` imports from modules with enums only.
6. [Added](https://github.com/neutron-org/neutron-sdk/blob/c89d40468bc2f06493b94f0ed058b28c794f574a/proto-build/src/transformers.rs#L624-L628) `#[allow(dead_code)]` attribute to the corresponding `***Querier` structs.
7. [Added](https://github.com/neutron-org/neutron-sdk/blob/c89d40468bc2f06493b94f0ed058b28c794f574a/proto-build/src/transformers.rs#L587-L591) `#[allow(clippy::too_many_arguments)]` attribute to the corresponding `***Querier` structs functions.
8. Regenerated schema.

### CODE REVIEW

File: packages/neutron-std-derive/tests/query.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+)]
+#[proto_message(type_url = "/osmosis.tokenfactory.v1beta1.QueryDenomsFromCreatorResponse")]
+pub struct QueryDenomsFromCreatorResponse {
+    #[prost(string, repeated, tag = "1")]
+    pub denoms: ::prost::alloc::vec::Vec<::prost::alloc::string::String>,
+}
+
+fn main() {
+    let _: QueryRequest<Empty> = QueryDenomsFromCreatorRequest {
+        creator: "osmo1sr9zm2pq3xrru7l7gz632t2rqs9caet9xulwvapcqagq9pytkcgqwfc3nk".to_string(),
```
Comment (pr0n00gler) [inline packages/neutron-std-derive/tests/query.rs:27]: Must be neutron address, please check it everywhere

File: packages/neutron-std-derive/tests/struct.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    pub subdenom: ::prost::alloc::string::String,
+}
+
+fn main() {
+    assert_eq!(
+        MsgCreateDenom::TYPE_URL,
+        "/osmosis.tokenfactory.v1beta1.MsgCreateDenom"
+    );
+    let msg = MsgCreateDenom {
+        sender: "osmo1sr9zm2pq3xrru7l7gz632t2rqs9caet9xulwvapcqagq9pytkcgqwfc3nk".to_string(),
```
Comment (pr0n00gler) [inline packages/neutron-std-derive/tests/struct.rs:20]: Must be neutron address, please check it everywhere

File: proto-build/src/code_generator.rs [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+                    .find(|e| {
+                        e.file_name()
+                            .to_str()
+                            .map(|s| s == "buf.yaml" || s == "buf.yml")
+                            .unwrap_or(false)
+                    })
+                    .map(|e| e.path().parent().unwrap().to_path_buf())
+                    .unwrap()
+            };
+            // let buf_root = WalkDir::new(&self.root.join(&project.project_dir))
```
Comment (pr0n00gler) [inline proto-build/src/code_generator.rs:143]: Commented code, remove it please, if it's not needed

File: proto-build/src/main.rs [hunk 4]
```diff
@@ -1,338 +1,173 @@
...
-        .unwrap_or_else(|e| match e.kind() {
-            io::ErrorKind::NotFound => panic!(
-                "error running '{:?}': command not found. Is it installed?",
-                cmd.as_ref()
-            ),
-            _ => panic!("error running '{:?}': {:?}", cmd.as_ref(), e),
-        });
+/// The ics23 commit or tag to be cloned and used to build the proto files
+///
+/// cosmos-sdk deps for `osmo/v0.47` is `v0.9.0` but that has no buf.yml,
```
Comment (pr0n00gler) [inline proto-build/src/main.rs:39]: We have Cosmos SDK 0.50, please this issue is still present. If it's not, remove comment, please

File: packages/neutron-std-derive/Cargo.toml
Comment (pr0n00gler) [inline packages/neutron-std-derive/Cargo.toml:1]: Please leave a comment, that this package a modified copy of the original osmosis package and attach a link to it

File: proto-build/Cargo.toml
Comment (pr0n00gler) [inline proto-build/Cargo.toml:1]: Please attach a comment to the original osmosis repo

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

File: packages/neutron-std-derive/tests/query.rs (hunk 1 above)
Comment (joldie777) [reply packages/neutron-std-derive/tests/query.rs]: Done

File: packages/neutron-std-derive/tests/struct.rs (hunk 2 above)
Comment (joldie777) [reply packages/neutron-std-derive/tests/struct.rs]: Done

File: proto-build/src/code_generator.rs (hunk 3 above)
Comment (joldie777) [reply proto-build/src/code_generator.rs]: Removed

File: proto-build/src/main.rs (hunk 4 above)
Comment (joldie777) [reply proto-build/src/main.rs]: Done

File: packages/neutron-std-derive/Cargo.toml
Comment (joldie777) [reply packages/neutron-std-derive/Cargo.toml:1]: Done

File: proto-build/Cargo.toml
Comment (joldie777) [reply proto-build/Cargo.toml:1]: Done

File: Makefile [hunk 5]
```diff
@@ -30,4 +30,4 @@ check_contracts:
 build: schema clippy test fmt doc compile check_contracts
 
 build-proto:
-	@cargo run --bin proto-build $(revision)
+	@cargo run --bin proto-build
```
Comment (pr0n00gler) [inline Makefile:33]: Please update README accordingly

Comment (joldie777) [reply Makefile:33]: Done

**Review by pr0n00gler — APPROVED** [review]

**Review by NeverHappened — APPROVED** [review]

**Pull request merged**